Formatting, wording, and link updates, add section on "Standard Names" (#57)
NCAR/ccpp-doc: CCPPtechnical/source/CompliantPhysicsParams.rst

dependent on scientific innovation, demonstrated value, and compliance with the rules
described below. The criteria for acceptance of a scheme into the CCPP is under development.

> It is recommended that parameterizations be partitioned into the smallest units that will be used independently.

Reviewer: I do not agree with the change from comprised to partitioned. Perhaps processes should be partitioned into parameterizations but  parameterization should not be partitioned. If this is not clear, or not good English, let's think of another way to express it.
```suggestion
It is recommended that parameterizations be comprised of the smallest units that will be used independently.
```

General Rules
=============
A CCPP-compliant scheme is written in the form of Fortran modules. Each scheme must be in its own module, and must include at least one of the

> following subroutines (*entry points*): *_timestep_init*, *_init*, *_run*, *_finalize*,

Author: Done

.. _scheme_template:
.. literalinclude:: ./_static/scheme_template.F90
   :language: fortran
   :lines: 10-48

*Listing 2.1: Fortran template for a CCPP-compliant scheme showing the _run subroutine. The structure for the other phases (*\ _timestep_init, _init, _finalize, *and* _timestep_finalize\ *) is identical.*

The three lines in the example template beginning ``!> \section`` are required. They begin with `!` and so will be treated as comments by the Fortran compiler, but are interpreted by Doxygen 
as part of the process to create scientific documentation. Those lines specifically insert an external file containing metadata
information (in this case, ``scheme_template_run.html``) in the documentation. See more on this topic in
:numref:`Section %s <SciDoc>`.

All external information required by the scheme must be passed in via the argument list, including physical constants. Statements
such as ``use EXTERNAL_MODULE`` should not be used for passing in any data.
See :numref:`Section %s <UsingConstants>` for more information on
how to use physical constants.

Note that standard names, variable names, module names, scheme names and subroutine names are all case insensitive.

> Interstitial modules (*schemename_pre* and *schemename_post*) can be included in a scheme's module if any part of the physics

Reviewer 2: I don't think that this is the right interpretation of the replaced text. I think that the original text just says that a developer can create *schemename_pre* and/or *schemename_post* schemes should the scheme *schemename* need any scheme-specific calculation performed before or after the scheme, assuming that there is a good reason for such code NOT to be included in the scheme itself, like if the interstitial code is host-specific or otherwise only needed when the scheme is part of a specific suite.

Reviewer 2: Geez, that was a long sentence. Let me know if it doesn't make sense.

Author: Thanks, I think I understand. I was confused as to why a different subroutine would be needed sequentially before and/or after the main scheme, but being host-dependent makes sense. I've updated the wording.

Standard names
==============

Variables available for CCPP physics schemes are identified by their unique *standard name*. This
policy is in place to ensure that variables will always be unique and unambiguous when communicating
between different schemes and different host models. Schemes are free to use their own variable
names within their individual codes, but these variables must be assigned to a *standard name* within
the scheme's metadata table as described in :numref:`Section %s <IOVariableRules>`.

Standard names are listed and defined in a GitHub repository (https://github.com/ESCOMP/CCPPStandardNames),
along with rules for adding new standard names as needed. While an effort is made to comply with 
existing *standard name* definitions of the Climate and Forecast (CF) conventions (http://cfconventions.org), 
additional names are used in the CCPP to cover the wide range of use cases the CCPP intends to include.
Each hash of the CCPP Physics repository contains information in the top-level ``README.md`` file
indicating which version of the CCPPStandardNames repository corresponds to that version of CCPP code.

> An up-to-date list of available standard names for each host model can be found by running the CCPP *prebuild* script (described in :numref:`Chapter %s <CCPPPreBuild>`), which will generate a LaTeX source file that can be compiled to produce a PDF file with all variables 

Reviewer: ```suggestion
An up-to-date list of available standard names for a given host model can be found by running the CCPP *prebuild* script (described in :numref:`Chapter %s <CCPPPreBuild>`), which will generate a LaTeX source file that can be compiled to produce a PDF file with all variables 
```

defined by the host model and requested by the physics schemes.

.. _IOVariableRules:

Input/Output Variable (argument) Rules
======================================

* A ``standard_name`` cannot be assigned to more than one local variable (``local_name``).
  The ``local_name`` of a variable can be chosen freely and does not have to match the
  ``local_name`` in the host model.

* All variable information (standard_name, units, dimensions) must match the specifications on
  the host model side, but sub-slices can be used/added in the host model. For example, when

>   using the UFS Weather Model as the host model, tendencies are split in ``GFS_typedefs.meta``

Reviewer: We use UFS Atmosphere in various parts of the document, referring to fv3atm. This is because CCPP is not used in the ocean, waves, or sea ice parts of the UFS Weather Model. I suggest reverting this change.

Standard names
==============

Variables available for CCPP physics schemes are identified by their unique *standard name*. This
policy is in place to ensure that variables will always be unique and unambiguous when communicating
between different schemes and different host models. Schemes are free to use their own variable
names within their individual codes, but these variables must be assigned to a *standard name* within
the scheme's metadata table as described in :numref:`Section %s <IOVariableRules>`.

Standard names are listed and defined in a GitHub repository (https://github.com/ESCOMP/CCPPStandardNames),
along with rules for adding new standard names as needed. While an effort is made to comply with 
existing *standard name* definitions of the Climate and Forecast (CF) conventions (http://cfconventions.org), 
additional names are used in the CCPP to cover the wide range of use cases the CCPP intends to include.
Each hash of the CCPP Physics repository contains information in the top-level ``README.md`` file
indicating which version of the CCPPStandardNames repository corresponds to that version of CCPP code.

> A list of available standard names for each host model can be found in 

Author: I hope someone can clarify for me whether these documents should still be referenced; are they up-to-date and consistent with the CCPPStandardNames repository?

Reviewer 2: This is a good question. These are DEFINITELY not up-to-date. Since up-to-date versions can be generated by running ccpp_prebuild.py with one of the hosts (as explained in the text), I would argue that is sufficient. Otherwise, someone will need to regenerate them using the release code and push them to the repository.

Author: Thanks for the clarification. I've updated the wording to just have users generate their own PDFs
